integrations/iron/src/main/scala/sttp/iron/codec/iron/TapirCodecIron.scala

```diff
@@ -0,0 +1,232 @@
+package sttp.tapir.codec.iron
+
+import sttp.tapir.Schema
+import io.github.iltotore.iron.Constraint
+import io.github.iltotore.iron.:|
+import io.github.iltotore.iron.refineEither
+import io.github.iltotore.iron.refineOption
+import io.github.iltotore.iron.constraint.any.*
+import io.github.iltotore.iron.constraint.string.*
+import io.github.iltotore.iron.constraint.collection.*
+import io.github.iltotore.iron.constraint.numeric.*
+
+import sttp.tapir.CodecFormat
+import sttp.tapir.Codec
+import sttp.tapir.DecodeResult
+import sttp.tapir.Validator
+import sttp.tapir.ValidationResult
+import scala.reflect.ClassTag
+import sttp.tapir.ValidationError
+import io.github.iltotore.iron.constraint.any.Not
+import io.github.iltotore.iron.macros.union.IsUnion
+import io.github.iltotore.iron.macros.intersection.IsIntersection
+
+import scala.compiletime.*
+import scala.util.NotGiven
+
+import sttp.tapir.typelevel.IntersectionTypeMirror
```

Comment:
**nitpick:** Could you reorganize imports to have this section with other `sttps`?

Reply:
Sure! did my best to organize them manually and remove unused ones